lib/shopify_app/controller_concerns/login_protection.rb

```diff
@@ -116,6 +116,21 @@ def login_url_params(top_level:)
       query_params
     end
 
+    def return_to_relative_path
+      relative_path(session[:return_to]) || relative_path(params[:return_to])
+    end
+
+    def relative_path(url)
```

Comment:
This logic seems like it would do unexpected things. For example if I set the URI to `https://myotherdomain.com/login` it'll get rewritten to `/login`. An alternative would be to check if the parsed URI is relative, and to set the path to `/` if not. The only snag here would be the situation where it's equal to the request host (and therefore safe) but still fully qualified. The code would look something like:

```ruby
uri = URI.parse(url)
if uri.relative?
  uri.to_s
else
  '/'
end
```